xgi/drawing/draw.py

```diff
@@ -1707,111 +1771,154 @@ def draw_dihypergraph(
-            node_fc_cmap=edge_marker_fc_cmap,
-            zorder=2,
-            params=settings,
-            rescale_sizes=rescale_sizes,
+    if hyperedge_labels is not None:
+        draw_node_labels(
+            H.dual(),
+            pos=edge_pos,
+            node_labels=hyperedge_labels,
+            ax_nodes=ax,
+            zorder=3,
         )
 
-        if hyperedge_labels:
-            labels = {i: ii for i, ii in zip(phantom_nodes, H_conv.edges)}
-            draw_node_labels(
-                HG_phantom,
-                pos=pos,
-                node_labels=labels,
-                ax_nodes=ax,
-                font_color_nodes="white",
-                zorder=3,
-            )
+    # check validity of input values
+    if np.any(dyad_lw < 0):
+        raise ValueError("dyad_lw cannot contain negative values.")
 
-    else:
-        phantom_node_collection = None
+    # interpolate if needed
+    if rescale_sizes and isinstance(dyad_lw, np.ndarray):
+        dyad_lw = _interp_draw_arg(
+            dyad_lw, settings["min_dyad_lw"], settings["max_dyad_lw"]
+        )
+
+    try:
+        dyad_collection = _draw_directed_edges(
+            edge_ids,
+            ax,
+            DH,
+            settings,
+            node_pos,
+            edge_pos,
+            dyads_c_mapped,
+            dyad_lw=dyad_lw,
+            dyad_color=dyad_color,
+            arrowsize=arrowsize,
+            arrowstyle=arrowstyle,
+            connectionstyle=connectionstyle,
+            node_size=node_size,
+            node_shape=node_shape,
+            edge_marker_size=edge_marker_size,
+            edge_marker_shape="s",
+        )
+    except UnboundLocalError as e:
+        dyads = to_bipartite_edgelist(H)
+        dyad_pos = np.asarray(
+            [(node_pos[e[0]], edge_pos[e[1]]) for e in dyads if e[1] in edge_ids]
+        )
+
+        dyad_collection = LineCollection(
+            dyad_pos,
+            colors=dyad_color,
+            linewidths=dyad_lw,
+            antialiaseds=(1,),
+            cmap=settings["dyad_color_cmap"],
+            zorder=0,
+        )
+
+        ax.add_collection(dyad_collection)
 
-    for dyad in H_conv.edges.filterby("size", 2).members():
-        try:
-            index = max(n for n in G_aug.nodes if isinstance(n, int)) + 1
-        except ValueError:
-            # The list of node-labels has no integers, so I start from 0
-            index = 0
-        G_aug.add_edges_from([[list(dyad)[0], index], [list(dyad)[1], index]])
+    pos = {}
+    for i, n in enumerate(node_pos):
+        pos[i] = node_pos[n]
+    n = H.num_nodes
+    for i, e in enumerate(edge_pos):
+        pos[i + n] = edge_pos[e]
 
-    edges = DH.edges.filterby("size", 2, "geq")
+    # compute axis limits
+    _update_lims(pos, ax)
 
-    nodes_id = list(DH.nodes)  # in original order, to find index for arrays
+    return ax, (node_collection, edge_marker_collection, dyad_collection)
+
+
+def _draw_directed_edges(
+    edge_ids,
```

Comment:
maybe second after ax, don't remember what we do in other functions

Reply:
Fixed. I deleted this arg and added `max_order` as an arg.